File: Gridly-Client/src/app/Utils/icon.factory.ts
import {IconModel} from "../Models/Icon.Model";

export function MapIconData(iconModel?: IconModel) : IconModel{
  return {
    name: iconModel?.name ?? "",
    type: iconModel?.type ?? "",
    base64Data: iconModel?.base64Data ?? "",
  } as IconModel;
}
MapIconData.Override = function(override: Partial<IconModel>, iconModel?: IconModel): IconModel {
  return MapIconData({
    name: override.name ??  iconModel?.name ?? "",
    type: override?.type ?? iconModel?.type ?? "",
    base64Data: override?.base64Data ?? iconModel?.base64Data ?? "",
  } as IconModel);
};

File: Factories/IconFactory.cs
using Gridly.Entities;
using Gridly.Models;

namespace Gridly.Factories
{
    public static class IconFactory
    {
        public static IconModel Create(IconModel? icon)
       => new IconModel
       {
           Name = icon?.Name ?? string.Empty,
           Type = icon?.Type ?? string.Empty,
           Base64Data = icon?.Base64Data ?? string.Empty,
           MaterialIcon = icon?.MaterialIcon ?? "box"
       };

        public static IconModel Create(IconEntity entity)
       => new IconModel
       {
           Id = entity.Id,
           Name = entity.Name ?? string.Empty,
           Type = entity.Type ?? string.Empty,
           Base64Data = entity.Base64Data ?? string.Empty,
           MaterialIcon = entity.MaterialIcon ?? string.Empty
       };
    }
}
